src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts

```diff
@@ -725,63 +745,116 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
+        } else if (signals) {
+            const objects = await this.globalStorageClient.listObjectsInCollection<EARDataSubmissionItem>(namespace);
+            const newSubmissionValues = this.getNewEARSubmissionValues(signals, objects, "APPROVED");
+            const recipients = await this.getEARRecipientUsers(signals, modules);
+
+            const message = this.getEARNotificationText(signals, modules, "approved");
+            this.sendNotifications(message, message, [], recipients);
+
+            return await this.globalStorageClient.saveObject<EARDataSubmissionItem[]>(namespace, newSubmissionValues);
+        }
+    }
+
+    private async approveByModule(modules: GLASSDataSubmissionModule[], items: GLASSDataSubmissionItemIdentifier[]) {
+        const selectedModule = modules.find(module => module.id === _.first(items)?.module)?.name ?? "";
+        const { AMC, AMR, AMR_FUNGHI, AMR_INDIVIDUAL, EGASP } = moduleMapping;
+
+        switch (selectedModule) {
+            case AMC: {
+                const amcPrograms = modules.find(module => module.name === AMC)?.programs ?? [];
+                const amcProductRegisterProgramId = await this.getAMCProductRegisterId(); // the AMC - Product Register program is tne only tracker program in this module
+
+                const amcEventPrograms = _.filter(
+                    amcPrograms,
+                    amcProgram => amcProgram.id !== amcProductRegisterProgramId
+                );
+                const amcTrackerPrograms = _.filter(
+                    amcPrograms,
+                    amcProgram => amcProgram.id === amcProductRegisterProgramId
+                );
+
+                _.forEach(amcEventPrograms, async amcProgram => await this.duplicateEventProgram(amcProgram, items));
+                _.forEach(
+                    amcTrackerPrograms,
+                    async amcProgram => await this.duplicateTrackerProgram(amcProgram, items)
+                );
+                break;
             }
-            if (module === "AMR") {
-                const amrDataSets = modules.find(module => module.name === "AMR")?.dataSets ?? [];
-                const amrQuestionnaires = modules.find(module => module.name === "AMR")?.questionnaires ?? [];
+            case AMR: {
+                const amrDataSets = modules.find(module => module.name === AMR)?.dataSets ?? [];
+                const amrQuestionnaires = modules.find(module => module.name === AMR)?.questionnaires ?? [];
 
                 _.forEach(amrDataSets, async amrDataSet => await this.duplicateDataSet(amrDataSet, items));
                 _.forEach(
                     amrQuestionnaires,
                     async amrQuestionnaire => await this.duplicateDataSet(amrQuestionnaire, items)
                 );
+                break;
             }
-            if (module === "AMR - Fungal") {
+            case AMR_FUNGHI: {
```

Comment:
good to change this to AMR_FUNGAL to ensure uniformity of name. Not a blocker.

Reply:
i use AMR_FUNGHI because the tracker/trackedEntities api returns the AMR_Fungal module as AMR_FUNGHI and i am also making use of the `moduleMapping` to get tracked entities